tests/log_util_test.py

```diff
@@ -0,0 +1,336 @@
+# Copyright 2014 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import inspect
+import os
+import subprocess
+import sys
+import threading
+import unittest
+
+from perfkitbenchmarker import log_util
+from perfkitbenchmarker import vm_util
+
+TEMP_DIR = '/tmp/perfkitbenchmarker/log_util_test'
+STDERR_PATH = TEMP_DIR + '/stderr.log'
+LOG_PATH = TEMP_DIR + '/pkb.log'
+FILE_NAME = (
+    os.path.split(inspect.getframeinfo(inspect.currentframe()).filename))[-1]
+
+
+def RunCommand(cmd):
+  p = subprocess.Popen(cmd)
+  p.wait()
+
+
+class LogUtilTestCase(unittest.TestCase):
```

Comment:
I'm not sure the whole logging system needs to be exercised here, and threading makes this complicated.
Recommend unit testing `ThreadLogContext` and `PkbLogFilter` directly.